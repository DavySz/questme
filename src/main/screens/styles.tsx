import { GestureHandlerRootView } from "react-native-gesture-handler";
import styled from "styled-components/native";

export const GestureWrapper = styled(GestureHandlerRootView)`
  flex: 1;
`;
