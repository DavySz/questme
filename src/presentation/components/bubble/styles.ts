import styled, { css } from "styled-components/native";
import { ICircleStyles, IPositionWrapperStyles } from "./types";

export const getWrapperPosition = (
  position: Partial<IPositionWrapperStyles>
) => {
  const customPosition = [];

  if (position.bottom) {
    customPosition.push(css`
      bottom: ${position.bottom}px;
    `);
  }

  if (position.left) {
    customPosition.push(css`
      left: ${position.left}px;
    `);
  }

  if (position.top) {
    customPosition.push(css`
      top: ${position.top}px;
    `);
  }

  if (position.right) {
    customPosition.push(css`
      right: ${position.right}px;
    `);
  }

  return customPosition;
};

export const CircleFill = styled.View<ICircleStyles>`
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;

  border-radius: ${({ size }) => size / 2}px;

  background-color: ${({ theme }) => theme.colors.support["accent-2"]};

  opacity: 0.1;
`;

export const CircleOutline = styled.View<ICircleStyles>`
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;

  border-radius: ${({ size }) => size / 2}px;

  background-color: ${({ theme }) => theme.colors.transparent};
  border: 1px solid ${({ theme }) => theme.colors.support["accent-2"]};

  opacity: 0.1;
`;

export const PositionWrapper = styled.View<Partial<IPositionWrapperStyles>>`
  position: absolute;
  ${(props) => getWrapperPosition(props as Partial<IPositionWrapperStyles>)};
`;
