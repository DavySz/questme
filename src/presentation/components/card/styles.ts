import styled from "styled-components/native";

export const Wrapper = styled.View`
  flex-direction: row;

  align-items: center;
  justify-content: space-between;

  width: 100%;
  padding: 8px;

  border-radius: 20px;
  border: 2px solid ${({ theme }) => theme.colors.neutral["grey-5"]};

  background-color: ${({ theme }) => theme.colors.neutral.white};
`;

export const Content = styled.View`
  gap: 6px;
`;

export const Group = styled.View`
  gap: 6px;
`;
