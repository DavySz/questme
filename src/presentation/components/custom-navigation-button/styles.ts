import styled from "styled-components/native";

export const Wrapper = styled.TouchableOpacity`
  border-radius: 24px;
  height: 48px;
  width: 48px;

  justify-content: center;
  align-items: center;

  margin-top: -20px;
`;

export const Container = styled.View`
  border-radius: 32px;
  height: 64px;
  width: 64px;

  justify-content: center;
  align-items: center;

  background-color: ${({ theme }) => theme.colors.transparent};

  margin-top: -20px;
`;
