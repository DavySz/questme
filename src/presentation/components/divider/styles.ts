import styled from "styled-components/native";

export const Line = styled.View`
  width: 100%;
  height: 1px;
  flex-direction: row;
  background-color: ${({ theme }) => theme.colors.neutral["gey-4"]};
  margin: 0 16px;
`;
