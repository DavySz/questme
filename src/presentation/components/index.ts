export * from "./button";
export * from "./text";
export * from "./screen-template";
export * from "./divider";
