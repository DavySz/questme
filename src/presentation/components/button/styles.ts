import styled, { css } from "styled-components/native";
import {
  IButtonIconStyles,
  IRootStyle,
  TButtonVariant,
  TIconSide,
} from "./types";
import { TTheme, TThemeColors } from "@presentation/styles";

export const getButtonBackground = (
  variant: TButtonVariant,
  theme: TTheme
): string => {
  const colors: Record<TButtonVariant, string> = {
    primary: theme.colors.primary,
    secondary: theme.colors.neutral["grey-5"],
    tertiary: theme.colors.transparent,
    facebook: theme.colors.company.facebook,
    google: theme.colors.neutral.white,
    location: theme.colors.neutral.white,
  };

  return colors[variant];
};

export const getButtonTextColor = (variant: TButtonVariant): TThemeColors => {
  const color: Record<TButtonVariant, TThemeColors> = {
    primary: "neutral-white",
    secondary: "primary",
    tertiary: "primary",
    facebook: "neutral-white",
    google: "neutral-black",
    location: "primary",
  };

  return color[variant] as TThemeColors;
};

const isFull = (isFull: boolean) => {
  if (isFull) {
    return css`
      width: 100%;
    `;
  }
};

const getIconSide = (side: TIconSide) => {
  if (side === "left") {
    return css`
      margin-right: 16px;
    `;
  }

  if (side === "right") {
    return css`
      margin-left: 16px;
    `;
  }

  return null;
};

const getBorder = (variant: TButtonVariant, theme: TTheme) => {
  if (variant === "google") {
    return css`
      border: 1px solid ${theme.colors.neutral["grey-4"]};
    `;
  }
};

export const Button = styled.TouchableOpacity.attrs({
  activeOpacity: 0.7,
})<IRootStyle>`
  ${({ full }) => isFull(full)};
  ${({ variant, theme }) => getBorder(variant, theme as TTheme)};

  background-color: ${({ theme, variant }) =>
    getButtonBackground(variant, theme as TTheme)};

  border-radius: 20px;
  padding: 16px;

  flex-direction: row;

  align-items: center;
  justify-content: center;
`;

export const IconWrapper = styled.View<IButtonIconStyles>`
  ${({ side }) => getIconSide(side)}
`;
