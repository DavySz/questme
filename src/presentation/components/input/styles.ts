import styled from "styled-components/native";
import { IInputWrapperStyles } from "./types";
import { TTheme } from "@presentation/styles";

const getBorderColor = (isFocused: boolean, theme: TTheme) => {
  if (isFocused) return theme.colors.primary;
  return theme.colors.neutral["grey-5"];
};

export const CustomInput = styled.TextInput`
  width: 100%;
  font-family: ${({ theme }) => theme.fonts.regular};
  font-size: 16px;
`;

export const InputWrapper = styled.View<IInputWrapperStyles>`
  flex-direction: row;

  width: 100%;
  padding: 16px;
  border-radius: 20px;

  gap: 16px;

  border: 2px solid
    ${({ isFocused, theme }) => getBorderColor(isFocused, theme as TTheme)};

  background-color: ${({ theme }) => theme.colors.neutral.white};
`;

export const Wrapper = styled.View`
  margin-bottom: 8px;
`;
