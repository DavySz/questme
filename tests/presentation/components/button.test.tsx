import { Button } from "@presentation/components";
import GoogleIcon from "@presentation/assets/icons/google-icon.svg";
import { customRender } from "../../utils/custom-render";
import { theme } from "@presentation/styles";

describe("Button", () => {
  describe("Button.Root", () => {
    test("Should render Button.Root without errors", () => {
      const tree = customRender(
        <Button.Root variant="google">
          <Button.Text>any-text</Button.Text>
        </Button.Root>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("Button.Text", () => {
    test("Should render Button.Text without errors", () => {
      customRender(
        <Button.Root variant="google">
          <Button.Text>any-text</Button.Text>
        </Button.Root>
      );
    });

    test("Should render correct children text", () => {
      const { getByText } = customRender(
        <Button.Root variant="google">
          <Button.Text>any-text</Button.Text>
        </Button.Root>
      );

      expect(getByText("any-text")).toBeTruthy();
    });
  });

  describe("Button.Icon", () => {
    test("Should render Button.Icon without errors", () => {
      customRender(<Button.Icon icon={GoogleIcon} side="left" />);
    });

    test("Should render Button.Icon with color", async () => {
      const { getByTestId } = customRender(
        <Button.Icon icon={GoogleIcon} side="left" color="company-facebook" />
      );

      expect(getByTestId("icon-with-color")).toBeTruthy();
    });

    test("Should render Button.Icon without color", async () => {
      const { getByTestId } = customRender(
        <Button.Icon icon={GoogleIcon} side="left" />
      );

      expect(getByTestId("icon-without-color")).toBeTruthy();
    });

    test("Should add margin right of 16px when Button.Icon receive left prop", async () => {
      const { getByTestId } = customRender(
        <Button.Icon icon={GoogleIcon} side="left" />
      );

      const iconWrapper = getByTestId("icon-wrapper");
      const iconWrapperStyle = iconWrapper.props.style;

      expect(iconWrapperStyle.marginRight).toBe(16);
    });

    test("Should add margin left of 16px when Button.Icon receive right prop", async () => {
      const { getByTestId } = customRender(
        <Button.Icon icon={GoogleIcon} side="right" />
      );

      const iconWrapper = getByTestId("icon-wrapper");
      const iconWrapperStyle = iconWrapper.props.style;

      expect(iconWrapperStyle.marginLeft).toBe(16);
    });

    test("Should not add margin when Button.Icon receive none prop", async () => {
      const { getByTestId } = customRender(
        <Button.Icon icon={GoogleIcon} side="none" />
      );

      const iconWrapper = getByTestId("icon-wrapper");
      const iconWrapperStyle = iconWrapper.props.style;

      expect(iconWrapperStyle.marginRight).toBeUndefined();
      expect(iconWrapperStyle.marginLeft).toBeUndefined();
    });
  });

  describe("Button.*", () => {
    test("Should render all button parts without errors", () => {
      const { getByText } = customRender(
        <Button.Root variant="google">
          <Button.Text>any-text</Button.Text>
          <Button.Icon icon={GoogleIcon} side="right" />
        </Button.Root>
      );

      expect(getByText("any-text")).toBeTruthy();
    });

    test("Should render button with correct background color primary variant by default", () => {
      const { getByTestId } = customRender(
        <Button.Root testID="button-root-wrapper">
          <Button.Icon icon={GoogleIcon} side="right" />
        </Button.Root>
      );

      const button = getByTestId("button-root-wrapper");
      const buttonStyle = button.props.style;

      expect(buttonStyle.backgroundColor).toBe(theme.colors.primary);
    });

    test("Should render button with correct background color when variant is passed", () => {
      const { getByTestId } = customRender(
        <Button.Root variant="tertiary" testID="button-root-wrapper">
          <Button.Icon icon={GoogleIcon} side="right" />
        </Button.Root>
      );

      const button = getByTestId("button-root-wrapper");
      const buttonStyle = button.props.style;

      expect(buttonStyle.backgroundColor).toBe(theme.colors.transparent);
    });

    test("Should render button with 100% width if full prop is passed", () => {
      const { getByTestId } = customRender(
        <Button.Root testID="button-root-wrapper">
          <Button.Icon icon={GoogleIcon} side="right" />
        </Button.Root>
      );

      const button = getByTestId("button-root-wrapper");
      const buttonStyle = button.props.style;

      expect(buttonStyle.width).toBe("100%");
    });

    test("Should render button with content width if full prop is not passed", () => {
      const { getByTestId } = customRender(
        <Button.Root full={false} testID="button-root-wrapper">
          <Button.Icon icon={GoogleIcon} side="right" />
        </Button.Root>
      );

      const button = getByTestId("button-root-wrapper");
      const buttonStyle = button.props.style;

      expect(buttonStyle.width).toBeUndefined();
    });
  });
});
