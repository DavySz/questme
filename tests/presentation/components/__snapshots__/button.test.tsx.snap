// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button.Root Should render Button.Root without errors 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderColor": "#E6E6E6",
      "borderStyle": "solid",
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
      "width": "100%",
    }
  }
>
  <Text
    color="neutral-black"
    style={
      {
        "color": "#0C092A",
        "fontFamily": "Rubik_500Medium",
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    variant="body-normal-medium"
  >
    any-text
  </Text>
</View>
`;
